src/components/Tasks/TaskList.tsx

```diff
@@ -9,31 +9,61 @@ import * as EteSync from 'etesync';
 
 import { List } from '../../widgets/List';
 
-import { TaskType, PimType } from '../../pim-types';
+import { TaskType, PimType, TaskTags } from '../../pim-types';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
 import Checkbox from '@material-ui/core/Checkbox';
 import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import { Theme, withTheme } from '@material-ui/core/styles';
+
+import { useSelector } from 'react-redux';
 
 import TaskListItem from './TaskListItem';
 import QuickAdd from './QuickAdd';
+import Sidebar from './Sidebar';
+
+import { StoreState } from '../../store';
 
 const sortSelector = createSelector(
   (entries: TaskType[]) => entries,
   (entries) => entries.sort((a, b) => a.title.localeCompare(b.title))
 );
 
+const tagFilters = Object.assign(
+  {},
+  ...TaskTags.map((tag) => ({
+    [tag]: (x: TaskType) => x.tags.includes(tag),
+  }))
+);
+
+const filters = {
+  all: () => true,
+  ...tagFilters,
+};
+
 interface PropsType {
   entries: TaskType[];
   collections: EteSync.CollectionInfo[];
   onItemClick: (entry: TaskType) => void;
   onItemSave: (item: PimType, journalUid: string, originalItem?: PimType) => Promise<void>;
+  theme: Theme;
 }
 
-export default React.memo(function TaskList(props: PropsType) {
+export default React.memo(withTheme(function TaskList(props: PropsType) {
   const [showCompleted, setShowCompleted] = React.useState(false);
-  const entries = props.entries.filter((x) => showCompleted || !x.finished);
+  const settings = useSelector((state: StoreState) => state.settings.taskSettings);
+  const { filterBy } = settings;
+
+  const entries = props.entries.filter((x) => showCompleted || !x.finished).filter(filters[filterBy]);
```

Comment:
See below how I suggested to change it to `potentialEntries` and `entries`. If you do that, you can also do something like:
```
let entries = potentialEntries;
if (filterBy !== 'all') { // Or better yet, just have 'all' be null -> so filter by nothing.
    entries = entries.filter((x) => x.tags.includes(filterBy));
}
```

Then you can delete lines 32-43, you are more efficient in the no-filter case (the most common one) and the code is much clearer and shorter.

Reply:
This is much cleaner, and it will work well for filtering by tag. But I know that I also want to add the ability to filter by due date, so I can't just necessarily use `x.tags.includes(filterBy)`. That is why I have all the nonsense with creating an object of filter functions. Do you have a suggestion for a way I can handle this but still make it general enough to have any kind of filter functions?

Reply:
This would have been problematic anyway, because tags called "Today" or "Recent" or whatever wouldn't have worked...
I think these are just two different concepts that should maybe be in two different fields? So the settings could be:
```
enum FilterView { ... }

filterBy: {
   tag?: string;
   filter?: FilterView
}
```
Then you can just have them as two separate `if`s that filter one after the other? (and probably have one disable the other when switching between them if you don't want them to work in tandem).

Reply:
Ok, I took a slightly different approach. I didn't want to muck up `filterBy` by making it an object with multiple options. Instead, I append "tag" to any `filterBy` option that filters by tag. So for instance, if I select the "Home" tag, the `filterBy` option will be `"tag Home"`. Then I just check to see if the `filterBy` option starts with `"tag"` and I filter appropriately.

It got rid of that nasty `filters` object, and should allow things to stay pretty general.

Reply:
Looks good to me. I'm going to take a look at the whole patch again now.

Reply:
Awesome, thank you